test/common/event/dispatcher_impl_test.cc

```diff
@@ -33,6 +33,30 @@ using testing::Return;
 namespace Envoy {
 namespace Event {
 namespace {
+class Destroyable : public DeferredDeletable, public DispatcherThreadDeletable {
```

Comment:
Please avoid mixing DeferredDeletable and DispatcherThreadDeletable in the relevant tests.

The requirement for DispatcherThreadDeletable is that you schedule the delete from outside the dispatcher thread and verify that deletion correctly happens in the next iteration of the dispatcher loop, on shutdown and on dispatcher deletion.

Reply:
I will add the case that deleteInDispatcherThread raised the schedule.

However, the Destroyable here is to confirm the shutdown behavior. 
Destroying in any DeferDetetable object, DispatcherThreadDeletable object or postCallback could create another object. Thus DispatcherImpl::shutdown() needs to go over the 3 lists repeatedly. This loop in shutdown guarantees the objects are destroyed gracefully.

Reply:
Delete the test using the `Destroyable` here. 

The DispatcherThreadDeletable object scenarios are tested by individual test cases